bob-common/src/mapper.rs

```diff
@@ -121,6 +121,47 @@ impl Virtual {
         self.vdisks.get(&id).expect("vdisk not found").nodes()
     }
 
+    fn find_support_node_index(
+        nodes: &[Node],
+        target_nodes: &[Node],
+        offset: usize,
+    ) -> SupportIndexResult {
+        let mut avail = 0;
+        for i in 0..nodes.len() {
+            let node = &nodes[i];
+            if node.connection_available() && target_nodes.iter().all(|n| n.index() != node.index())
+            {
+                if avail == offset {
+                    return SupportIndexResult {
+                        index: Some(i),
+                        avail: avail + 1,
+                    };
+                }
+                avail += 1;
+            }
+        }
+        return SupportIndexResult { index: None, avail };
+    }
+
+    fn find_support_node_offset(nodes: &[Node], target_nodes: &[Node], offset: usize) -> usize {
```

Comment:
Write the comment, that this function looks for an offset in the list of nodes, respecting the uniform distribution of resulting nodes

Reply:
Done